<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>671</width>
    <height>450</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="tabShape">
   <enum>QTabWidget::Triangular</enum>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QWidget" name="verticalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>10</y>
      <width>391</width>
      <height>51</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <item>
        <widget class="QPushButton" name="camera_info_button">
         <property name="text">
          <string>相机参数</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="construct_button">
         <property name="text">
          <string>光谱重构</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="spectral_curve_button">
         <property name="text">
          <string>光谱曲线</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QLabel" name="camera_image_label">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>375</width>
      <height>300</height>
     </rect>
    </property>
    
    <property name="text">
     <string/>
    </property>
   </widget>
   <widget class="Line" name="line_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>90</y>
      <width>400</width>
      <height>5</height>
     </rect>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>390</y>
      <width>400</width>
      <height>5</height>
     </rect>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_5">
    <property name="geometry">
     <rect>
      <x>400</x>
      <y>90</y>
      <width>20</width>
      <height>301</height>
     </rect>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_6">
    <property name="geometry">
     <rect>
      <x>0</x>
      <y>90</y>
      <width>20</width>
      <height>301</height>
     </rect>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Plain</enum>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="Line" name="line_7">
    <property name="geometry">
     <rect>
      <x>430</x>
      <y>0</y>
      <width>20</width>
      <height>421</height>
     </rect>
    </property>
    <property name="frameShadow">
     <enum>QFrame::Raised</enum>
    </property>
    <property name="lineWidth">
     <number>2</number>
    </property>
    <property name="orientation">
     <enum>Qt::Vertical</enum>
    </property>
   </widget>
   <widget class="QPlainTextEdit" name="info_text_edit">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>320</y>
      <width>211</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>20</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <item>
      <widget class="QPushButton" name="openCameraButton">
       <property name="text">
        <string>打开相机</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>50</y>
      <width>196</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <item>
      <widget class="QPushButton" name="open_com_button">
       <property name="text">
        <string>打开串口</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="comboBox">
       <item>
        <property name="text">
         <string>/dev/ttyUSB0</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>/dev/ttyUSB1</string>
        </property>
       </item>
       <item>
        <property name="text">
         <string>/dev/ttyUSB2</string>
        </property>
       </item>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_3">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>110</y>
      <width>171</width>
      <height>28</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <item>
      <widget class="QPushButton" name="rot_button">
       <property name="text">
        <string>旋转</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="dir_checkbox">
       <property name="text">
        <string>正/反</string>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
   <widget class="QPushButton" name="save_button">
    <property name="geometry">
     <rect>
      <x>170</x>
      <y>400</y>
      <width>80</width>
      <height>26</height>
     </rect>
    </property>
    <property name="text">
     <string>保存</string>
    </property>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_4">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>140</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <property name="spacing">
      <number>1</number>
     </property>
     <item>
      <widget class="QLabel" name="angle_label">
       <property name="text">
        <string>旋转角度：</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="angle_line_edit"/>
     </item>
    </layout>
   </widget>
   <widget class="QWidget" name="horizontalLayoutWidget_5">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>170</y>
      <width>171</width>
      <height>31</height>
     </rect>
    </property>
    <layout class="QHBoxLayout" name="horizontalLayout_6">
     <item>
      <widget class="QPushButton" name="auto_button">
       <property name="text">
        <string>自动拍照</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QComboBox" name="auto_box"/>
     </item>
    </layout>
   </widget>
   <widget class="Line" name="line_2">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>90</y>
      <width>231</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="Line" name="line_8">
    <property name="geometry">
     <rect>
      <x>440</x>
      <y>210</y>
      <width>231</width>
      <height>16</height>
     </rect>
    </property>
    <property name="orientation">
     <enum>Qt::Horizontal</enum>
    </property>
   </widget>
   <widget class="QWidget" name="verticalLayoutWidget_2">
    <property name="geometry">
     <rect>
      <x>450</x>
      <y>230</y>
      <width>211</width>
      <height>80</height>
     </rect>
    </property>
    <layout class="QVBoxLayout" name="verticalLayout_2">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_9">
       <item>
        <widget class="QLabel" name="camera_label">
         <property name="text">
          <string>相机状态：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="camera_status">
         <property name="text">
          <string>关闭</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_8">
       <item>
        <widget class="QLabel" name="com_label">
         <property name="text">
          <string>串口状态：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="com_status">
         <property name="text">
          <string>关闭</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <item>
        <widget class="QLabel" name="zhengfan_label">
         <property name="text">
          <string>旋转方向：</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="zhengfan_stauts">
         <property name="text">
          <string>正</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
